### ISSUE TITLE
Twitter sign in usign OAuthProvider

### ISSUE THREAD

**Issue by GuanacoDevs** [firebase/firebase-android-sdk#645]
I have been using Twitter-kit since I couldn't find a way to use Twitter Sign In without it.
Until today that checking for the appropriate steps to login with Twitter found out that you have updated the guide for [Twitter Auth](https://firebase.google.com/docs/auth/android/twitter-login?authuser=0), and that's just great!!! Is so convenient and allows me to get rid of Twitter-kit

However after implementation and successful sign in with Twitter I can't recover control of the app, I have to manually close the chrome tab.

- If it is the first time I'm trying to sign In with Twitter, after accepting the usage of the app the Custom chrome Tab shows an error that I can't reproduce, because the Chrome Tab keeps signin with the last user automatically, despite of having left unchecked the "remind me..." checkbox, but after closing the chrome tab with the error I'm already signed in with Twitter.

- If I logout and then try to login again, it will redirect me to last signed in user, so it won't let me change user for the Twitter login. After successful login the chrome tab gets closed but the app is out of focus, cleared only by taping the screen.

Is there anything I can do or is the expected behavior?, it is not user friendly at the moment but is quite developer friendly.

Perhaps you could add in the documentation for users that comes from using Twitter Kit that `FirebaseAuth.getInstance().startActivityForSignInWithProvider` already returns a `Firebase.AuthResult`, it took me awhile since I was use to Twitter-kit.

**Comment by samtstern** [comment]
cc @malcolmdeck @lsirac

**Comment by lsirac** [comment]
Hi GuanacoDevs,

I cannot reproduce the second issue where the app is out of focus after a successful login. 

I did reproduce the first issue which is unfortunately on Twitters end. If you want to reproduce an error you can log out of Twitter manually. I also can't find any custom parameters to force a reauth - their documentation is pretty limited.

**Comment by GuanacoDevs** [comment]
@lsirac I have uploaded [this video](https://photos.app.goo.gl/GmHb8eusBYDvr6yc8) and below is the summary of the issues in the timeline.

As you suggested I went to Chrome first to logout from my account.
Then at:

**00:18**   I start the login process to Twitter

**00:50**   Login has been sucessful but the chrome tab won't close displaying the message
- "This page does not exists or has expired. Try to login..."
- Then I have to tap the screen and close the chrome Tab to get back to the application
- Then I logout to login again

**01:14**   I restart the login process

**01:28**   Login has been sucessful(but it won't let me choose account, as you pointed out I have to logout from Chrome if I want to change account), the chrome tab closes, and shows a progress dialog

**01:31**   The application is still on the background, I have to tap to get focus on the application

As a developer this is great as it allows me to get rid of Twitter-Kit and before I was offering Twitter signin only if the device had the application, but it doesn't look like user friendly as a regular user will not know what is going on that he/she has to tap and close to get back to the application.

I'm not using FirebaseUI and this device is a Xperia Z5 Premium with 7.1.1

**Comment by lsirac** [comment]
Thanks for the video!

Have you tried reproducing these issues on different devices?

**Comment by GuanacoDevs** [comment]
@lsirac Not really, I just have tried with two different emulators 10 && 8 only and my device.
Same result on each

**Comment by lsirac** [comment]
Hey GuanacoDevs, are you still having this issue?

**Comment by GuanacoDevs** [comment]
Did anything changed?
I just have tried on the same app, which is currently in production, and the same behavior.
I'm currently adding the same login feature to another app, and while in debug, I noticed that if I cancelled the twitter signing, instead of proceeding, forwards me to a page were I can see a button "Return to..." and after pressing it goes back to my app as expected. I tried the same in the app of which I opened this question and is not working the same.
[Here is a video](https://photos.app.goo.gl/HpBVpABRRBHJR7cv8) for that. Unfortunately I can't test the full signing in this app as I'm working in the migration of data in that one, I'll report back when finished.
